diff --git a/lib/conditions.js b/lib/conditions.js
index e884c98..a19cc35 100644
--- a/lib/conditions.js
+++ b/lib/conditions.js
@@ -28,7 +28,7 @@ module.exports.paramMatches = function (params) {
     return conditionalChecker(params, function (matches, value) {
         var result;
         if (_.isArray(matches)) {
-            result = _.contains(matches, value);
+            result = _.includes(matches, value);
         } else {
             result = _.isEqual(matches, value);
         }
diff --git a/lib/validators.js b/lib/validators.js
index b466e29..f8a6097 100644
--- a/lib/validators.js
+++ b/lib/validators.js
@@ -101,7 +101,7 @@ var validators = module.exports = {
                     }
                     return validateKey.error;
 
-                }, false, this);
+                }.bind(this), false);
                 // Merge renamed elements
                 _.extend(submittedValue, renamedElements);
                 if (keyError) {
@@ -149,7 +149,7 @@ function validateCollectionValues(context, collection, validationRules) {
 
 
 var notSupported = ['len', 'notNull', 'isNull', 'notEmpty'];
-var validatorFunctionKeys = _.difference(_.functions(validator), notSupported);
+var validatorFunctionKeys = _.difference(_.functionsIn(validator), notSupported);
 
 _.each(validatorFunctionKeys, function (element, index, array) {
     if (!_.has(validators, element)) {
