diff --git anndata/_io/h5ad.py anndata/_io/h5ad.py
index 2fb34946..198f41b9 100644
--- anndata/_io/h5ad.py
+++ anndata/_io/h5ad.py
@@ -7,6 +7,11 @@ from types import MappingProxyType
 from typing import Callable, Type, TypeVar, Union
 from typing import Collection, Sequence, Mapping
 
+try:
+    from importlib.metadata import version
+except ImportError:
+    from importlib_metadata import version
+
 import h5py
 import numpy as np
 import pandas as pd
@@ -17,7 +22,12 @@ from .._core.sparse_dataset import SparseDataset
 from .._core.file_backing import AnnDataFileManager
 from .._core.anndata import AnnData
 from .._core.raw import Raw
-from ..compat import _from_fixed_length_strings, _clean_uns, Literal
+from ..compat import (
+    _from_fixed_length_strings,
+    _decode_structured_array,
+    _clean_uns,
+    Literal,
+)
 from .utils import (
     report_read_key_on_error,
     report_write_key_on_error,
@@ -28,6 +38,9 @@ from .utils import (
     EncodingVersions,
 )
 
+# For allowing h5py v3
+# https://github.com/theislab/anndata/issues/442
+H5PY_V3 = version("h5py") >= "3"
 
 H5Group = Union[h5py.Group, h5py.File]
 H5Dataset = Union[h5py.Dataset]
@@ -451,7 +464,14 @@ def _read_raw(
 @report_read_key_on_error
 def read_dataframe_legacy(dataset) -> pd.DataFrame:
     """Read pre-anndata 0.7 dataframes."""
-    df = pd.DataFrame(_from_fixed_length_strings(dataset[()]))
+    if H5PY_V3:
+        df = pd.DataFrame(
+            _decode_structured_array(
+                _from_fixed_length_strings(dataset[()]), dtype=dataset.dtype
+            )
+        )
+    else:
+        df = pd.DataFrame(_from_fixed_length_strings(dataset[()]))
     df.set_index(df.columns[0], inplace=True)
     return df
 
@@ -478,7 +498,7 @@ def read_series(dataset) -> Union[np.ndarray, pd.Categorical]:
         categories = dataset.attrs["categories"]
         if isinstance(categories, h5py.Reference):
             categories_dset = dataset.parent[dataset.attrs["categories"]]
-            categories = categories_dset[...]
+            categories = read_dataset(categories_dset)
             ordered = bool(categories_dset.attrs.get("ordered", False))
         else:
             # TODO: remove this code at some point post 0.7
@@ -489,9 +509,11 @@ def read_series(dataset) -> Union[np.ndarray, pd.Categorical]:
                 "AnnData. Rewrite the file ensure you can read it in the future.",
                 FutureWarning,
             )
-        return pd.Categorical.from_codes(dataset[...], categories, ordered=ordered)
+        return pd.Categorical.from_codes(
+            read_dataset(dataset), categories, ordered=ordered
+        )
     else:
-        return dataset[...]
+        return read_dataset(dataset)
 
 
 # @report_read_key_on_error
@@ -527,6 +549,10 @@ def read_group(group: h5py.Group) -> Union[dict, pd.DataFrame, sparse.spmatrix]:
 @read_attribute.register(h5py.Dataset)
 @report_read_key_on_error
 def read_dataset(dataset: h5py.Dataset):
+    if H5PY_V3:
+        string_dtype = h5py.check_string_dtype(dataset.dtype)
+        if (string_dtype is not None) and (string_dtype.encoding == "utf-8"):
+            dataset = dataset.asstr()
     value = dataset[()]
     if not hasattr(value, "dtype"):
         return value
@@ -539,7 +565,10 @@ def read_dataset(dataset: h5py.Dataset):
             return value[0]
     elif len(value.dtype.descr) > 1:  # Compound dtype
         # For backwards compat, now strings are written as variable length
+        dtype = value.dtype
         value = _from_fixed_length_strings(value)
+        if H5PY_V3:
+            value = _decode_structured_array(value, dtype=dtype)
     if value.shape == ():
         value = value[()]
     return value
diff --git anndata/compat/__init__.py anndata/compat/__init__.py
index 16e92945..3e3f94c6 100644
--- anndata/compat/__init__.py
+++ anndata/compat/__init__.py
@@ -1,8 +1,9 @@
 from copy import deepcopy
 from functools import reduce
-from typing import Union, Mapping, MutableMapping
+from typing import Union, Mapping, MutableMapping, Optional
 from warnings import warn
 
+import h5py
 from scipy.sparse import spmatrix
 import numpy as np
 import pandas as pd
@@ -84,6 +85,34 @@ def _from_fixed_length_strings(value):
     return value.astype(new_dtype)
 
 
+def _decode_structured_array(
+    arr: np.ndarray, dtype: Optional[np.dtype] = None, copy: bool = False
+) -> np.ndarray:
+    """
+    h5py 3.0 now reads all strings as bytes. There is a helper method which can convert these to strings,
+    but there isn't anything for fields of structured dtypes.
+
+    Params
+    ------
+    arr
+        An array with structured dtype
+    dtype
+        dtype of the array. This is checked for h5py string data types.
+        Passing this is allowed for cases where array may have been processed by another function before hand.
+    """
+    if copy:
+        arr = arr.copy()
+    if dtype is None:
+        dtype = arr.dtype
+    # codecs.decode is 2x slower than this lambda, go figure
+    decode = np.frompyfunc(lambda x: x.decode("utf-8"), 1, 1)
+    for k, (dt, _) in dtype.fields.items():
+        check = h5py.check_string_dtype(dt)
+        if check is not None and check.encoding == "utf-8":
+            decode(arr[k], out=arr[k])
+    return arr
+
+
 def _to_fixed_length_strings(value: np.ndarray) -> np.ndarray:
     """\
     Convert variable length strings to fixed length.
