diff --git a/package/config.js b/package/config.js
index 2c5b56c..e91cea5 100644
--- a/package/config.js
+++ b/package/config.js
@@ -1,5 +1,5 @@
 const { resolve } = require('path')
-const { load } = require('js-yaml')
+const { load, CORE_SCHEMA, mergeTag } = require('js-yaml')
 const { readFileSync } = require('fs')
 const { merge } = require('webpack-merge')
 const { ensureTrailingSlash } = require('./utils/helpers')
@@ -9,12 +9,12 @@ const configPath = require('./configPath')
 const defaultConfigPath = require.resolve('../lib/install/config/webpacker.yml')
 
 const getDefaultConfig = () => {
-  const defaultConfig = load(readFileSync(defaultConfigPath), 'utf8')
+  const defaultConfig = load(readFileSync(defaultConfigPath), { schema: CORE_SCHEMA.withTags(mergeTag) })
   return defaultConfig[railsEnv] || defaultConfig.production
 }
 
 const defaults = getDefaultConfig()
-const app = load(readFileSync(configPath), 'utf8')[railsEnv]
+const app = load(readFileSync(configPath), { schema: CORE_SCHEMA.withTags(mergeTag) })[railsEnv]
 
 const config = merge(defaults, app)
 config.outputPath = resolve(config.public_root_path, config.public_output_path)
