diff --git a/komposer/types/kubernetes.py b/komposer/types/kubernetes.py
index 3ccb98d..240faf3 100644
--- a/komposer/types/kubernetes.py
+++ b/komposer/types/kubernetes.py
@@ -8,7 +8,7 @@ from pathlib import Path
 from typing import Literal, Optional, Union
 
 from dotenv import dotenv_values
-from pydantic import Field
+from pydantic import Field, SerializeAsAny
 
 from komposer.types.base import CamelCaseImmutableBaseModel
 from komposer.types.cli import Context
@@ -259,4 +259,4 @@ class List(CamelCaseImmutableBaseModel):
     api_version: Literal["v1"] = "v1"
     kind: Literal["List"] = "List"
     # We should be able to use Field(..., discriminator="kind") here
-    items: list[Item] = []
+    items: list[SerializeAsAny[Item]] = []
