diff --git featuretools/primitives/standard/aggregation/num_unique.py featuretools/primitives/standard/aggregation/num_unique.py
index b0021e5d..dcba4818 100644
--- featuretools/primitives/standard/aggregation/num_unique.py
+++ featuretools/primitives/standard/aggregation/num_unique.py
@@ -11,8 +11,14 @@ dd = import_or_none("dask.dataframe")
 class NumUnique(AggregationPrimitive):
     """Determines the number of distinct values, ignoring `NaN` values.
 
+    Args:
+        use_string_for_pd_calc (bool): Determines if the string 'nunique' or the function
+            pd.Series.nunique is used for making the primitive calculation. Put in place to
+            account for the bug https://github.com/pandas-dev/pandas/issues/57317.
+            Defaults to using the string.
+
     Examples:
-        >>> num_unique = NumUnique()
+        >>> num_unique = NumUnique(use_string_for_pd_calc=False)
         >>> num_unique(['red', 'blue', 'green', 'yellow'])
         4
 
@@ -29,6 +35,9 @@ class NumUnique(AggregationPrimitive):
     compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
     description_template = "the number of unique elements in {}"
 
+    def __init__(self, use_string_for_pd_calc=True):
+        self.use_string_for_pd_calc = use_string_for_pd_calc
+
     def get_function(self, agg_type=Library.PANDAS):
         if agg_type == Library.DASK:
 
@@ -51,7 +60,6 @@ class NumUnique(AggregationPrimitive):
 
             return dd.Aggregation(self.name, chunk=chunk, agg=agg, finalize=finalize)
 
-        elif agg_type == Library.SPARK:
+        if self.use_string_for_pd_calc:
             return "nunique"
-
         return pd.Series.nunique
diff --git featuretools/computational_backends/feature_set_calculator.py featuretools/computational_backends/feature_set_calculator.py
index d67ce2f7..26837ca8 100644
--- featuretools/computational_backends/feature_set_calculator.py
+++ featuretools/computational_backends/feature_set_calculator.py
@@ -820,7 +820,6 @@ class FeatureSetCalculator(object):
                 # work)
                 if is_instance(base_frame, (dd, ps), "DataFrame"):
                     to_merge = base_frame.groupby(groupby_col).agg(to_agg)
-
                 else:
                     to_merge = base_frame.groupby(
                         base_frame[groupby_col],
