diff --git vcr/stubs/__init__.py vcr/stubs/__init__.py
index 4d4bb39..aaad91b 100644
--- vcr/stubs/__init__.py
+++ vcr/stubs/__init__.py
@@ -66,6 +66,7 @@ class VCRHTTPResponse(HTTPResponse):
         self.reason = recorded_response["status"]["message"]
         self.status = self.code = recorded_response["status"]["code"]
         self.version = None
+        self.version_string = None
         self._content = BytesIO(self.recorded_response["body"]["string"])
         self._closed = False
         self._original_response = self  # for requests.session.Session cookie extraction
