diff --git a/lib/merge.js b/lib/merge.js
--- a/lib/merge.js
+++ b/lib/merge.js
@@ -3,7 +3,7 @@
 exports = module.exports = getMergeConfig;
 
 function getMergeConfig(global, local) {
-    return _.merge(global, local, function (object, source) {
+    return _.mergeWith(global, local, function (object, source) {
         if (object && (object.data || object.code)) {
             console.error('Clashes presets: ', object, source);
             return object;
