diff --git a/index.js b/index.js
--- a/index.js
+++ b/index.js
@@ -68,7 +68,7 @@ function merge(opts, defaults, config, callbacks) {
         toMerge = commandLineArgs;
     }
 
-    var simpleMerged = _.merge(_.cloneDeep(defaults), toMerge, function (a, b) {
+    var simpleMerged = _.mergeWith(_.cloneDeep(defaults), toMerge, function (a, b) {
         return _.isArray(a) ? _.union(a, b) : undefined;
     });
 
