diff --git xarray/tests/test_duck_array_ops.py xarray/tests/test_duck_array_ops.py
index c29e9d74..afcf10ec 100644
--- xarray/tests/test_duck_array_ops.py
+++ xarray/tests/test_duck_array_ops.py
@@ -157,7 +157,7 @@ class TestOps:
         assert 1 == count(np.datetime64("2000-01-01"))
 
     def test_where_type_promotion(self):
-        result = where([True, False], [1, 2], ["a", "b"])
+        result = where(np.array([True, False]), np.array([1, 2]), np.array(["a", "b"]))
         assert_array_equal(result, np.array([1, "b"], dtype=object))
 
         result = where([True, False], np.array([1, 2], np.float32), np.nan)
@@ -214,7 +214,7 @@ class TestOps:
         assert_array_equal(result, np.array([1, "b"], dtype=object))
 
     def test_concatenate_type_promotion(self):
-        result = concatenate([[1], ["b"]])
+        result = concatenate([np.array([1]), np.array(["b"])])
         assert_array_equal(result, np.array([1, "b"], dtype=object))
 
     @pytest.mark.filterwarnings("error")
