diff --git a/index.js b/index.js
--- a/index.js
+++ b/index.js
@@ -10,6 +10,7 @@
 var url
 var req
 var coapOptionSeperator = ','
+var urlArg
 
 function collectOptions (val, memo) {
   var coapOptionRegex = new RegExp('\\d{1}\\' + coapOptionSeperator + '\\w+')
@@ -40,20 +41,29 @@
 
 ;['GET', 'PUT', 'POST', 'DELETE'].forEach(function (name) {
   program
-    .command(name.toLowerCase())
+    .command(name.toLowerCase() + ' <url>')
     .description('performs a ' + name + ' request')
-    .action(function () { method = name })
+    .action(function (target) {
+      method = name
+      urlArg = target
+    })
 })
 
+program
+  .arguments('[url]')
+  .action(function (target) {
+    urlArg = target
+  })
+
 program.parse(process.argv)
 
-if (!program.args[0]) {
+if (!urlArg) {
   program.outputHelp()
   process.exit(-1)
 }
 
 try {
-  url = new URL.URL(decodeURIComponent(program.args[0]))
+  url = new URL.URL(decodeURIComponent(urlArg))
 } catch (err) {
   if (err instanceof TypeError) {
     console.log('Invalid URL. Protocol is not given or URL is malformed.')
